webserver/webserver/src/main/java/io/helidon/webserver/WebTracingConfig.java

```diff
@@ -224,13 +226,24 @@ public void accept(ServerRequest req, ServerResponse res) {
 
     static final class RequestSpanHandler implements Handler {
         private static final String TRACING_SPAN_HTTP_REQUEST = "HTTP Request";
+        private final AtomicBoolean checkedIfShouldTrace = new AtomicBoolean();
+        private volatile boolean shouldTrace = true;
 
         RequestSpanHandler() {
         }
 
         @Override
         public void accept(ServerRequest req, ServerResponse res) {
-            doAccept(req, res);
+            if (checkedIfShouldTrace.compareAndSet(false, true)) {
```

Comment:
In case there is no transition of `shouldTrace` back to `true` we can save few ops by 
```java
if (shouldTrace && checkedIfShouldTrace.compareAndSet(false, true)) {
```

Reply:
But its almost un-measurable

Reply:
Fixed